File: sdks/ruby/examples/child/simple_fanout.rb
# frozen_string_literal: true

require "hatchet-sdk"
require_relative "worker"

hatchet = Hatchet::Client.new

# > Running a task from within a task
SPAWN_TASK = hatchet.task(name: "SpawnTask") do |input, ctx|
  result = CHILD_TASK_WF.run({ "message" => "Hello, World!" })
  { "results" => result }
end
# !!

File: sdks/python/examples/child/simple-fanout.py
from typing import Any

from examples.child.worker import SimpleInput, child_task
from hatchet_sdk.context.context import Context
from hatchet_sdk.hatchet import Hatchet

hatchet = Hatchet()


# > Running a Task from within a Task
@hatchet.task(name="SpawnTask")
async def spawn(input: None, ctx: Context) -> dict[str, Any]:
    # Simply run the task with the input we received
    result = await child_task.aio_run(
        input=SimpleInput(message="Hello, World!"),
    )

    return {"results": result}


# !!
